ext/simplexml/simplexml.c

```diff
@@ -457,7 +457,7 @@ static zval *sxe_prop_dim_write(zend_object *object, zval *member, zval *value,
 			}
 
 			if (!Z_STRLEN_P(member)) {
-				php_error_docref(NULL, E_WARNING, "Cannot write or create unnamed %s", attribs ? "attribute" : "element");
+				zend_value_error("Cannot create or write %s with an empty name", attribs ? "attributes" : "elements");
```

Comment:
This error message reads a bit awkward to me. I think it would be fine to make it just `Cannot create attribute with an empty name`, as such an attribute cannot already exist, so the "or write" terminology doesn't seem strictly necessary.